pkg/controller/registry/resolver/step_resolver.go

```diff
@@ -31,16 +32,43 @@ type StepResolver interface {
 }
 
 type OperatorStepResolver struct {
-	subLister              v1alpha1listers.SubscriptionLister
-	csvLister              v1alpha1listers.ClusterServiceVersionLister
-	client                 versioned.Interface
-	globalCatalogNamespace string
-	resolver               *Resolver
-	log                    logrus.FieldLogger
+	subLister                 v1alpha1listers.SubscriptionLister
+	csvLister                 v1alpha1listers.ClusterServiceVersionLister
+	client                    versioned.Interface
+	globalCatalogNamespace    string
+	resolver                  *Resolver
+	log                       logrus.FieldLogger
+	globalCatalogSourceToggle GlobalNamespaceToggle
 }
 
 var _ StepResolver = &OperatorStepResolver{}
 
+type GlobalNamespaceToggle interface {
+	ShouldEnableGlobalNamespaceResolution(ctx context.Context, namespace string) (bool, error)
+}
+
+func NewNamespaceAnnotationGlobalCatalogToggle(opClient operatorclient.ClientInterface) *NamespaceAnnotationGlobalCatalogToggle {
+	return &NamespaceAnnotationGlobalCatalogToggle{
+		opClient: opClient,
+	}
+}
+
+type NamespaceAnnotationGlobalCatalogToggle struct {
+	opClient operatorclient.ClientInterface
+}
+
+func (n *NamespaceAnnotationGlobalCatalogToggle) ShouldEnableGlobalNamespaceResolution(ctx context.Context, namespace string) (bool, error) {
```

Comment:
Is it safer to have this be `ShouldDisableGlobalNamespaceResolution` and preserve the default resolution behavior by default? I think we should see this as users can opt-out of global resolution on a per-namespace basis, versus opting-in to global catalog resolution, which is the existing behavior